### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by Neojume**
- GetAlertsActor now has a timeout after which it will return the accumulated results.
- LogParserActor now has mutable fields instead of parameters that are passed around in the receive method.

### CODE REVIEW

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 1]
```diff
@@ -20,55 +21,50 @@ class LogParserActor extends Actor with ActorLogging {
   this: AlertRuleActorCreator =>
   private type AlertRuleActorRef = ActorRef
 
-  def receive: Receive = normal(Nil, Map())
+  private var metrics = List[Metric]()
```
Comment (TimSoethout) [inline src/main/scala/performanceanalysis/LogParserActor.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/scala-academy/performance-analysis/file/2667153634/issues/source?bid=3220915&fileBranchId=3331880#l24)

File: src/main/scala/performanceanalysis/GetAlertsActor.scala [hunk 2]
```diff
@@ -3,30 +3,48 @@ package performanceanalysis
...
   */
 object GetAlertsActor {
   def props(ruleList: List[ActorRef], originalSender: ActorRef): Props = Props(new GetAlertsActor(ruleList, originalSender))
+
+  val timeOut = 4.seconds
 }
 
 class GetAlertsActor(alertActors: List[ActorRef], originalSender: ActorRef) extends Actor with ActorLogging {
 
+  import context.dispatcher
```
Comment (TimSoethout) [inline src/main/scala/performanceanalysis/GetAlertsActor.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/scala-academy/performance-analysis/file/2667153633/issues/source?bid=3220915&fileBranchId=3331880#l19)

Comment (TimSoethout) [inline src/main/scala/performanceanalysis/GetAlertsActor.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/scala-academy/performance-analysis/file/2667418563/issues/source?bid=3220915&fileBranchId=3331880#l19)

Comment (Neojume) [inline src/main/scala/performanceanalysis/GetAlertsActor.scala:19]: The context is only known inside the actor

File: src/test/scala/performanceanalysis/GetAlertsActorSpec.scala [hunk 3]
```diff
@@ -45,5 +47,27 @@ class GetAlertsActorSpec(testSystem: ActorSystem) extends ActorSpecBase(testSyst
       senderProbe.expectMsg(AllAlertRuleDetails(Set[AlertRule](alertRule1, alertRule2)))
       senderProbe.expectTerminated(ref)
     }
+
+    "send result back if after 2 seconds not every actor has responded" in {
```
Comment (Neojume) [inline src/test/scala/performanceanalysis/GetAlertsActorSpec.scala]: change to 5 seconds as is tested

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 4]
```diff
@@ -22,85 +24,77 @@ class LogParserActor extends Actor with ActorLogging {
...
 
-  def receive: Receive = normal(Nil, List.empty, Map())
+  private val metrics = mutable.MutableList[Metric]()
+  private val logLines = mutable.MutableList[String]()
+  private val alertsByMetricKey = MutableMap[MetricKey, List[AlertRuleActorRef]]()
 
-  def normal(metrics: List[Metric],
-             logLines: List[String],
-             alertsByMetricKey: Map[MetricKey, List[AlertRuleActorRef]]): Receive = {
+  def receive: Receive = {
```
Comment (TimSoethout) [inline src/main/scala/performanceanalysis/LogParserActor.scala]: Codacy Issue found: [Cyclomatic complexity of 11 exceeds max of 10](https://www.codacy.com/app/scala-academy/performance-analysis/file/2671056344/issues/source?bid=3220915&fileBranchId=3331880#l31)

**Comment by TimSoethout** [comment]
Please fix the imports

**Pull request merged**